packages/marketplace/src/utils/auth-route.ts

```diff
@@ -53,29 +53,43 @@ export function getDefaultPathByLoginType({
   clientLoginRouteMatch,
   isDeveloperFirstTimeLoginComplete,
   isClientFirstTimeLoginComplete,
+  isDesktopMode,
 }: {
   loginType: LoginType
   developerLoginRouteMatch: match<{}> | null
   clientLoginRouteMatch: match<{}> | null
   isDeveloperFirstTimeLoginComplete?: boolean
   isClientFirstTimeLoginComplete?: boolean
+  isDesktopMode?: boolean
 }) {
   switch (loginType) {
     case 'ADMIN':
       return Routes.ADMIN_APPROVALS
     case 'DEVELOPER':
       if (clientLoginRouteMatch) {
+        if (isDesktopMode) {
```

Comment:
@willmcvay The logic here very complicated. So could we do reject if user login wrong portal instead of redirect ?

Reply:
Currently, we let a developer account access to both routes "/developer/login" and  "/client/login" (and vice versa) although he's already logged in and then we will call a function named `getDefaultPathByLoginType` in `login.tsx` to get a URL to redirect them to an appropriate place.

This makes our logic in `getDefaultPathByLoginType`  becomes more complicated especially when we need to handle `isDesktopMode` case